central/imagev2/datastore/store/common/parts_test.go

```diff
@@ -0,0 +1,522 @@
+							ComponentId:          getTestComponentID(testComponents[1], imageID),
+						},
+					},
+					{
+						CVEV2: &storage.ImageCVEV2{
+							Id:        getTestCVEID(testCVEs["cve2comp1"], getTestComponentID(testComponents[1], imageID)),
+							ImageIdV2: imageID,
+							CveBaseInfo: &storage.CVEInfo{
+								Cve:       "cve2",
+								CreatedAt: ts,
+							},
+							NvdScoreVersion: storage.CvssScoreVersion_UNKNOWN_VERSION,
+							HasFixedBy: &storage.ImageCVEV2_FixedBy{
+								FixedBy: "ver3",
+							},
+							IsFixable:            true,
+							FirstImageOccurrence: ts,
+							ComponentId:          getTestComponentID(testComponents[1], imageID),
+						},
+					},
+				},
+			},
+			{
+				ComponentV2: &storage.ImageComponentV2{
+					Id:        getTestComponentID(testComponents[2], imageID),
+					Name:      "comp2",
+					Version:   "ver1",
+					ImageIdV2: imageID,
+					HasLayerIndex: &storage.ImageComponentV2_LayerIndex{
+						LayerIndex: 2,
+					},
+				},
+				Children: []CVEPartsV2{
+					{
+						CVEV2: &storage.ImageCVEV2{
+							Id:        getTestCVEID(testCVEs["cve1comp2"], getTestComponentID(testComponents[2], imageID)),
+							ImageIdV2: imageID,
+							CveBaseInfo: &storage.CVEInfo{
+								Cve:       "cve1",
+								CreatedAt: ts,
+							},
+							NvdScoreVersion: storage.CvssScoreVersion_UNKNOWN_VERSION,
+							HasFixedBy: &storage.ImageCVEV2_FixedBy{
+								FixedBy: "ver2",
+							},
+							IsFixable:            true,
+							FirstImageOccurrence: ts,
+							ComponentId:          getTestComponentID(testComponents[2], imageID),
+						},
+					},
+					{
+						CVEV2: &storage.ImageCVEV2{
+							Id:        getTestCVEID(testCVEs["cve2comp2"], getTestComponentID(testComponents[2], imageID)),
+							ImageIdV2: imageID,
+							CveBaseInfo: &storage.CVEInfo{
+								Cve:       "cve2",
+								CreatedAt: ts,
+							},
+							NvdScoreVersion:      storage.CvssScoreVersion_UNKNOWN_VERSION,
+							FirstImageOccurrence: ts,
+							ComponentId:          getTestComponentID(testComponents[2], imageID),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	splitActual, err := Split(imageV2, true)
+	assert.NoError(t, err)
+	protoassert.Equal(t, splitExpected.Image, splitActual.Image)
+
+	assert.Len(t, splitActual.Children, len(splitExpected.Children))
+	for i, expected := range splitExpected.Children {
+		actual := splitActual.Children[i]
+		protoassert.Equal(t, expected.ComponentV2, actual.ComponentV2)
+
+		assert.Len(t, actual.Children, len(expected.Children))
+		for i, e := range expected.Children {
+			a := actual.Children[i]
+			protoassert.Equal(t, e.CVEV2, a.CVEV2)
+		}
+	}
+
+	imageActual := Merge(splitActual)
+	protoassert.Equal(t, dedupedImageV2(imageID, imageName, imageSha), imageActual)
+}
+
+func getTestComponentID(testComponent *storage.EmbeddedImageScanComponent, imageID string) string {
+	id, _ := scancomponent.ComponentIDV2(testComponent, imageID)
+	return id
```

Comment:
why is error ignored?its better to  better to add require.NoError(t, err)

Reply:
Yeah good call. It was copied over from old image data model tests. I added the error assert.

Reply:
nice.  I like it.  I probably should have thought of it when I ignored that err before.